pkg/api/v2/reporter/reporter.go

```diff
@@ -272,74 +270,31 @@ func (r *reporter) WriteReports(ctx context.Context, resourceErrs ResourceReport
 		}
 
 		r.statusClient.SetStatus(resourceToWrite, status)
-		var updatedResource resources.Resource
 		writeErr := errors.RetryOnConflict(retry.DefaultBackoff, func() error {
-			var writeErr error
-			updatedResource, resourceToWrite, writeErr = r.attemptUpdateStatus(ctx, client, resourceToWrite, status)
-			return writeErr
+			return r.attemptUpdateStatus(ctx, client, resourceToWrite, status)
 		})
 
+		if errors.IsNotExist(writeErr) {
+			logger.Debugf("did not write report for %v : %v because resource was not found", resourceToWrite.GetMetadata().Ref(), status)
+			delete(resourceErrs, resource)
+			continue
+		}
+
 		if writeErr != nil {
```

Comment:
What is different about the 2 if statements here?  I am unfamiliar with `errors.IsNotExist(writeErr)`.

Reply:
that error is for the scenario when we try to patch a resource that no longer exists (e.g. a user deleted it). rather than report an error in the logs, we just remove it from our in memory report and swallow the non-error. for any other error type we will report the error